### PR TITLE
Benchmarks for Aderyn and individual detectors 

### PR DESCRIPTION
**Pull request by TilakMaddy**
Run `cargo bench` to see output like [this](https://app.warp.dev/block/ttZlrYP2pcKSPHAR7d9Kfk)

* every time you run the command it creates and stores reports in `target/criterion` (HTML files, charts, etc)
* in the CLI output itself you can see which detectors have increased in perf and which ones have regressed
* once you delete `target/criterion` you lose history

### CODE REVIEW

File: aderyn_driver/Cargo.toml [hunk 1]
```diff
@@ -10,4 +10,11 @@ license = "MIT"
 
 [dependencies]
 aderyn_core = { path  = "../aderyn_core", version = "0.0.19" }
-rayon = "1.8.0"
\ No newline at end of file
+rayon = "1.8.0"
+
+[dev-dependencies]
+criterion = "0.5.1"
+
```
Comment (TilakMaddy) [inline aderyn_driver/Cargo.toml:17]: This is the crate added as a dev dependency that helps us in benchmarking

**Comment by TilakMaddy** [comment]
@alexroan  Shall we add benchmarking template in nyth?

**Comment by TilakMaddy** [comment]
Draft because:

I want to add a benchmark for the overall aderyn process (where all detectors run in parallel),

not just for individual detectors sequentially (what we currently have)

**Comment by alexroan** [comment]
> I want to add a benchmark for the overall aderyn process (where all detectors run in parallel),
> 
> not just for individual detectors sequentially (what we currently have)

*Both* would be epic @TilakMaddy 👍

**Comment by alexroan** [comment]
> @alexroan Shall we add benchmarking template in nyth?

Not right now. I'm considering whether we actually need `Nyth` at all tbh.

**Comment by alexroan** [comment]
@TilakMaddy Do you still have TODOs on this one?

**Comment by TilakMaddy** [comment]
@alexroan I have added `cli/benchmarks.sh`. So now it does a couple of things

1) Copy the existing benchmarks (which I am attaching with this PR) into `target/`
2) Run the bench
3) Update the bench reports here (move them from `target/` to `benchmarks`)
4) Open the report in the browser to the main page where you can click on any of the options ( individual detector name (or) aderyn) performance and see how it has regressed/improved based on prev. No relying on CLI output

**Comment by TilakMaddy** [comment]
I'd prefer if CI could do this for every release and automatically commit the new reports to the codebase : )

**Review by alexroan — APPROVED** [review]
This scaffolding suffices for now. We'll do some work after this PR as part of #69 to add this to release flow.

**Pull request merged**